商品一覧表示機能

cc82tia opened the pull request:
※　売切表示除く

### what
-　出品されている商品一覧ページの作成
###why
-　フリマでの売買を行うにあたり、商品を見やすくするため

----------
ご確認とご指摘よろしくお願い致します

cc82tia committed:
商品一覧表示機能

売切表示除く

masayaomoto commented on app/controllers/items_controller.rb:
```diff
@@ -2,9 +2,8 @@ class ItemsController < ApplicationController
-    # @items = @items.includes(:user)
-    # .order("created_at DESC")
+    @items = Item.all
+    @items = @items.includes(:user).order("created_at DESC")
```
```
@items = Item.all
@items = @items.includes(:user).order("created_at DESC")
```
こちら```@items = Item.includes(:user).order("created_at DESC")```とすることで
1行で記述することができます。
修正しましょう！

cc82tia replied:
済

masayaomoto commented on app/views/items/index.html.erb:
```diff
@@ -146,19 +146,19 @@
+        <% end %>  
       </li>
       <%# //商品のインスタンス変数になにか入っている場合、中身のすべてを展開できるようにしましょう %>
 
```
167行目以下の商品のダミーが常時表示されてしまっているかとお見受けできます。
商品が存在する場合はダミーを表示しない条件分岐を記述しましょう！

cc82tia replied:
済

masayaomoto commented on app/views/items/show.html.erb:
```diff
@@ -7,7 +7,7 @@
       <%# <%= @item.name %> %>
```
こちらのファイルは商品詳細機能の実装範囲かと存じます。
こちらのプルリクエストでレビューをしてしまうと商品詳細機能のプルリクエストにて差分が確認できない可能性がございます。
現状としては、コメントアウトの状態に戻すようにお願いします。

cc82tia replied:
済

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

cc82tia committed:
商品一覧機能2

cc82tia commented:
※　売切表示除く

### what
-　出品されている商品一覧ページの作成
### why
-　フリマでの売買を行うにあたり、商品を見やすくするため

----------
ご確認とご指摘よろしくお願い致します

masayaomoto commented on app/views/items/index.html.erb:
```diff
@@ -146,41 +146,42 @@
-                <div class='star-btn'>
-                  <%= image_tag "star.png", class:"star-icon" %>
-                  <span class='star-count'>0</span>
+          <% if @item = nil %>
```
```<% if @item = nil %>```
こちらの条件ではダミー商品の表示・非表示がうまくできていないかと存じます。
@itemsのデータが無ければダミー商品を表示する条件に修正しましょう！

cc82tia replied:
https://gyazo.com/2c6c90ad69af191d84f23870822421cb

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

cc82tia committed:
商品一覧表示３

# what
-　出品されている商品一覧ページの作成

why
-　フリマでの売買を行うにあたり、商品を見やすくするため
--------
ご確認とご指摘よろしくお願い致します※　売切表示除く

NaturalRadio reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
